lib/resources/http.rb

```diff
@@ -0,0 +1,49 @@
+# encoding: utf-8
+# copyright: 2017, Criteo
+# author: Guilhem Lettron
+# license: All rights reserved
+
+require 'http'
+
+module Inspec::Resources
+  class Http < Inspec.resource(1)
+    name 'http'
+    desc 'Use the http InSpec audit resource to test http call.'
+    example "
+      describe http('http://localhost:8080/ping', auth: {user: 'user', pass: 'test'}, params: {format: 'html'}) do
+        its('status') { should cmp 200 }
+        its('body') { should cmp 'pong' }
+        its('Content-Type') { should cmp 'text/html' }
+      end
+    "
+
+    # rubocop:disable ParameterLists
+    def initialize(url, method: 'GET', params: nil, auth: {}, headers: {}, data: nil)
+      @url = url
+      @method = method
+      @params = params
+      @auth = auth
+      @headers = headers
+      @data = data
+      http = HTTP.headers(@headers)
+      http = http.basic_auth(@auth) unless @auth.empty?
+      @response = http.request(@method, @url, { body: @data, params: @params })
```

Comment:
We should do a lazy response request, otherwise the request will be done during resource instantiation. Lets have a private method `request` that is called by all other functions and it should also cache the result in `@response`